### PR TITLE
Updated listsinceblock rpc documentation

### LINKED ISSUE #8758: Fix documentations of rpc `listsinceblock`
## The second param of listsinceblock

From [issue 8752](https://github.com/bitcoin/bitcoin/issues/8752#issuecomment-247988691)

> The second parameter (target confirmation) is quite strange and should probably be better > documented in the RPC help of listsinceblock (check original comment: #199 (comment))

From the actual docs here is it is:

> target-confirmations:    (numeric, optional) The confirmations required, must be 1 or more

However, that's misleading at best. I've seen at least one project thinking that `target-confirmations` means "filter to only show transactions of n target-confirmations". It's a weird param but I'd document it as

> target-confirmations: (numeric, optional) Return the hash of the block that has target-confirmations of confirmations, for use in a further call to listsinceblock
## Transactions confirmations

Says:

> "confirmations": n,       (numeric) The number of confirmations for the transaction. Available for 'send' and 'receive' category of transactions.

It should also say when it's < 0, it means the transaction conflicted that many blocks ago
## bip125-replaceable

There's an undocumented field of transactions "bip125-replaceable" which I believe has three possibilities "yes" "no" and "unknown"  (under what circumstances it says unknown, should also be documented)

### PR DESCRIPTION
**Pull request by accraze**
This PR contains updates to the documentation for listsinceblock rpc.

fixes #8758

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1637,7 +1637,7 @@ UniValue listsinceblock(const JSONRPCRequest& request)
             "\nGet all transactions in blocks since block [blockhash], or all transactions if omitted\n"
             "\nArguments:\n"
             "1. \"blockhash\"   (string, optional) The block hash to list transactions since\n"
-            "2. target-confirmations:    (numeric, optional) The confirmations required, must be 1 or more\n"
+            "2. target-confirmations:    (numeric, optional) Return the hash of the block that has target-confirmations of confirmations, for use in a further call to listsinceblock.\n"
```
Comment (paveljanik) [inline src/wallet/rpcwallet.cpp]: This is surely wrong. It should describe the meaning of the argument. And it does so in the master.

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -1650,6 +1650,7 @@ UniValue listsinceblock(const JSONRPCRequest& request)
             "    \"vout\" : n,               (numeric) the vout value\n"
             "    \"fee\": x.xxx,             (numeric) The amount of the fee in " + CURRENCY_UNIT + ". This is negative and only available for the 'send' category of transactions.\n"
             "    \"confirmations\": n,       (numeric) The number of confirmations for the transaction. Available for 'send' and 'receive' category of transactions.\n"
+            "                                          When it's < 0, it means the transaction conflicted that many blocks ago.\n"
```
Comment (paveljanik) [inline src/wallet/rpcwallet.cpp]: But this is good!

**Comment by accraze** [comment]
@paveljanik I removed the incorrect description

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/9396/commits/3f67972654748a2f24e4aa95b6f3df5d4574ed55

**Comment by instagibbs** [comment]
nit: maybe add "e.g., -5 implies 5  blocks ago", in case someone reads it as "-5 blocks ago"(5 blocks in the future?)

ACK  https://github.com/bitcoin/bitcoin/pull/9396/commits/3f67972654748a2f24e4aa95b6f3df5d4574ed55

**Pull request merged**